apps/demo/src/app/checkout/components/place-order/place-order.component.ts

```diff
@@ -1,18 +1,26 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, Input, HostListener } from '@angular/core';
 import { Observable } from 'rxjs';
 import { Store, select } from '@ngrx/store';
 
 import * as fromDemoCheckout from '../../reducers';
-import { PlaceOrder } from '../../actions/checkout.actions';
+import { Cart } from '@daffodil/core';
+import { PlaceOrder } from '@daffodil/state';
 
 @Component({
   selector: 'demo-place-order',
   templateUrl: './place-order.component.html',
   styleUrls: ['./place-order.component.scss']
 })
-export class PlaceOrderComponent implements OnInit{
+export class PlaceOrderComponent implements OnInit {
 
+  @Input() cart: Cart;
   enablePlaceOrderButton$: Observable<boolean>;
+  
+  @HostListener('click', ['$event.target'])
+  placeOrder() {
+    this.store.dispatch(new PlaceOrder(this.cart));
+  }
+
 
   constructor(
```

Comment:
I'm not sure I understand why this component both accepts inputs and pulls content from State. This feels like an anti-pattern. It's like a "dumb-smart" or "mediocre" component. I think we should pick one or the other.

Reply:
I changed it to use selectors only